src/main/java/de/tum/in/www1/artemis/domain/settings/ide/UserIdeMappingId.java

```diff
@@ -22,27 +22,43 @@ public UserIdeMappingId(Long user, ProgrammingLanguage programmingLanguage) {
         this.programmingLanguage = programmingLanguage;
     }
 
-    @Override
-    public boolean equals(Object o) {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-
-        UserIdeMappingId that = (UserIdeMappingId) o;
-
-        if (!user.equals(that.user)) {
-            return false;
-        }
-        return programmingLanguage == that.programmingLanguage;
+    public Long getUser() {
+        return user;
     }
 
-    @Override
-    public int hashCode() {
-        int result = user.hashCode();
-        result = 31 * result + programmingLanguage.hashCode();
-        return result;
+    public ProgrammingLanguage getProgrammingLanguage() {
+        return programmingLanguage;
     }
+
+    public void setProgrammingLanguage(ProgrammingLanguage programmingLanguage) {
+        this.programmingLanguage = programmingLanguage;
+    }
+
+    public void setUser(Long user) {
+        this.user = user;
+    }
+
+    // @Override
+    // public boolean equals(Object o) {
+    // if (this == o) {
+    // return true;
+    // }
+    // if (o == null || getClass() != o.getClass()) {
+    // return false;
+    // }
+    //
+    // UserIdeMappingId that = (UserIdeMappingId) o;
+    //
+    // if (!user.equals(that.user)) {
+    // return false;
+    // }
+    // return programmingLanguage == that.programmingLanguage;
+    // }
+    //
+    // @Override
+    // public int hashCode() {
+    // int result = user.hashCode();
+    // result = 31 * result + programmingLanguage.hashCode();
+    // return result;
+    // }
```

Comment:
Don't forget to remove that code before merging - it does not really make sense to keep unused code (lava anti-pattern)

Reply:
i instantly pushed afterwards, cause i noticed too